event_substitution/wizard/wiz_event_substitution.py

```diff
@@ -73,6 +132,39 @@ def button_substitution_employee(self):
         self.holiday._update_presences_validate_holiday()
         self.holiday._update_partner_calendar_day(
             self.holiday, absence_type=self.holiday.holiday_status_id.id)
+        for registration in registrations:
+            wiz.with_context(
```

Comment:
¿Porque no los haces todos juntos y evitas el for?
`wiz.with_context({'active_ids': registrations.ids}).action_confirm_registrations()`